lib/rails/dom/testing/assertions/selector_assertions.rb

```diff
@@ -296,8 +298,10 @@ def nest_selection(selection)
           def nodeset(node)
             if node.is_a?(Nokogiri::XML::NodeSet)
               node
-            else
+            elsif node.present?
               Nokogiri::XML::NodeSet.new(node.document, [node])
+            else
+              raise MissingNodeError, 'No node/document found. Did you return a blank body in your response?'
```

Comment:
Instead of raising an exception we should just create a nodeset with an empty document, so all the assertions will return `false`.

Reply:
Yep I can do that. It won't resolve @brenogazzola feedback, but I'm not sure how much we want to couple this to the response code.

Reply:
I don't think we need to. His codebase failing to follow the redirect has nothing to do with the change we had in Rails or even the bug we are fixing here. His tests were just wrong, it always was. They are passing by mistake and isn't Rails job to be telling that he was asserting something that would always pass. With this change we will continue to have the tests passing like before.

If he deleted the view he was expecting to be rendered after the redirect the tests would also pass. Should Rails detect that? I don't think so.

Reply:
Alright, I pushed an alternative version.

The only issue left is if you do `assert_select count: 0` then that will still pass (whereas with the `raise` approach it always failed). Which is the example [here](https://github.com/rails/rails/pull/44632#issuecomment-1061052836). It's hard to know if it's "correct" or not. Technically it is correct (no nodes match) but not for the reasons you'd expect!

Reply:
`assert_select count: 0` that should pass. The body is empty, so there is nothing to select here.

Rails doesn't know the reason why the assertion is incorrect, there are many reasons why a body can be empty. It could be a 200 response with an empty body, and sometimes that is exactly what people are expecting.

Reply:
Yup. I think that it is better if the tests still pass. Just gave a suggestion based on the fact that the PR was about raising a message. 

Broken tests either done by us when we were not aware of assert_redirected_to, or changes in behavior which switched a 200 response to a 300